lib/FluxibleContext.js

```diff
@@ -190,46 +197,81 @@ FluxContext.prototype._initializeDispatcher = function initializeDispatcher() {
  */
-FluxContext.prototype.getActionContext = function getActionContext() {
-    if (this._actionContext) {
-        return this._actionContext;
-    }
+FluxContext.prototype.getActionContext = function getActionContext(meta) {
     var self = this;
 
-    if (!self._dispatcher) {
-        self._initializeDispatcher();
-    }
-
-    var actionContext = {
-        dispatch: this._dispatcher.dispatch.bind(this._dispatcher),
-        executeAction: this.executeAction.bind(this),
-        getStore: this._dispatcher.getStore.bind(this._dispatcher)
-    };
-
-    self._plugins.forEach(function pluginsEach(plugin) {
-        var actionContextPlugin = plugin.plugActionContext;
-        if (actionContextPlugin) {
-            actionContextPlugin(actionContext, self, self._app);
+    if (!self._actionContext) {
+        if (!self._dispatcher) {
+            self._initializeDispatcher();
         }
-    });
 
-    self._actionContext = actionContext;
+        var actionContext = {
+            dispatch: self._dispatcher.dispatch.bind(self._dispatcher),
+            executeAction: function actionExecuteAction (action, payload, callback) {
+                /*
+                 * Pass the current stack along.
+                 * If this action calls more actions,
+                 * we keep modifying and passing the stack along.
+                 */
+                var actionName = action.displayName || action.name;
+                // we have to clone the stack, can't just push to it
+                // since one action can execute multiple actions.
+                var nextStack = this.stack.concat([actionName]);
+                return self.executeAction(action, payload, callback, {
+                    nextStack: nextStack,
+                    id: this.id
+                });
+            },
+            getStore: self._dispatcher.getStore.bind(self._dispatcher)
+        };
+
+        self._plugins.forEach(function pluginsEach(plugin) {
+            var actionContextPlugin = plugin.plugActionContext;
+            if (actionContextPlugin) {
+                actionContextPlugin(actionContext, self, self._app);
+            }
+        });
+
+        self._actionContext = actionContext;
+    }
 
-    return self._actionContext;
+    var nextStack = (meta && meta.nextStack) || ['Action'];
+
+    /*
+     * We store the action's stack array on the `stack` property
+     * of the actionContext interface.
+     * You can access this in your actions with `context.stack`.
+     * Use the `displayName` property on your actions for better
+     * action tracing when your code gets minified in prod.
+     * One action can execute multiple actions, so we need to create
+     * a shallow clone with a new stack every time getActionContext is called
+     */
+    return objectAssign({}, self._actionContext, {
+        stack: nextStack,
+        id: (meta && meta.id) || generateUUID()
+    });
 };
 
+var tempThing = 0;
+function generateUUID () {
```

Comment:
For actions it would be rare to reach max int, but it may be possible. Perhaps it's safe to always use the randomized one from React but make sure it supports all browsers.